NEW add timespent API endpoints for projects and tasks add also cascading assignment of contacts to tasks

Jessica-creat opened the pull request:
## Description
Ajout de deux nouveaux endpoints API pour récupérer les temps passés sur les projets et tâches.

## Nouveaux Endpoints
- `GET /projects/{id}/timespent` - Récupère tous les temps passés sur l'ensemble des tâches d'un projet
- `GET /tasks/{id}/timespent` - Récupère les temps passés sur une tâche spécifique

## Tests effectués
- Projets avec multiples tâches et temps enregistrés

## Fichiers modifiés
- `htdocs/api/class/api_projects.class.php`
- `htdocs/api/class/api_tasks.class.php`

## Notes
Ces endpoints complètent l'API existante et suivent la même structure que les autres endpoints projets/tâches.

Jessica-creat committed:
demande d'ajout timespent

Jessica-creat committed:
Merge branch 'develop' of https://github.com/Dolibarr/dolibarr into develop

Jessica-creat committed:
feat(api): add timespent endpoints for projects and tasks

eldy committed:
Merge branch 'develop' into feature/api-timespent-endpoints

Jessica-creat committed:
add contact to product

eldy committed:
Merge branch 'develop' into feature/api-timespent-endpoints

Jessica-creat committed:
correction prettier

Jessica-creat committed:
revert

Jessica-creat committed:
feat(api): update timespent endpoints in api_projects and api_tasks

hregis commented:
@Jessica-creat and before here : 

`public function addToContact($id, $fk_socpeople, $type_contact, $source, $notrigger = 0, $affect_to_tasks = null)`

hregis commented:
@Jessica-creat and before here : 

`public function getTimespent($id)`

hregis commented:
@Jessica-creat and before here : 

`foreach ($this->project->lines as $task) {`

hregis commented:
@Jessica-creat après ça je t'invite à diner !

eldy committed:
Update api_projects.class.php

eldy committed:
Remove blank line before docblock in api_tasks.class.php

Removed unnecessary blank line before the docblock for the get contacts method.

eldy committed:
Merge branch 'develop' into feature/api-timespent-endpoints

eldy committed:
Update api_tasks.class.php

eldy committed:
Clarify return type in api_projects.class.php

Updated return type annotation for contact information array.

eldy committed:
Refine return type in get timespent method

Updated return type annotation for timespent method.

eldy committed:
Return raw contacts instead of cleaned data

eldy committed:
Update api_projects.class.php

eldy committed:
Return contacts array instead of cleaned data

Modified the return value to return the contacts array directly instead of cleaned data.
